framework/auth/views.py

```diff
@@ -49,26 +49,33 @@ def reset_password(auth, **kwargs):
         }
         raise HTTPError(400, data=error_data)
 
-    if request.method == 'POST' and form.validate():
-        # new random verification key, allows CAS to authenticate the user w/o password one time only.
-        user_obj.verification_key = security.random_string(20)
-        user_obj.set_password(form.password.data)
-        user_obj.save()
-        status.push_status_message('Password reset', kind='success', trust=False)
-        # Redirect to CAS and authenticate the user with a verification key.
-        return redirect(cas.get_login_url(
-            web_url_for('user_account', _absolute=True),
-            auto=True,
-            username=user_obj.username,
-            verification_key=user_obj.verification_key
-        ))
-
-    forms.push_errors_to_status(form.errors)
     return {
         'verification_key': verification_key,
     }
 
 
+def reset_password_post(verification_key):
+    """ attempt to reset password if request form passed
+    """
+    user_obj = get_user(verification_key=verification_key)
+    if not user_obj:
+        error_data = {
+            'message_short': 'Invalid url.',
+            'message_long': 'The verification key in the URL is invalid or has expired.'
+        }
+        raise HTTPError(400, data=error_data)
+
+    user_obj.verification_key = security.random_string(20)
+    try:
+        user_obj.set_password(request.json['password'])
+    except HTTPError as e:
```

Comment:
`set_password` should not raise an `HTTPError`; HTTP is outside of the responsibilities of the models. Instead, `set_password` should raise a `modularodm.exception.ValidationError` and that should be re-raised as an HTTPError here.